cpp/open3d/t/geometry/BoundingVolume.cpp

```diff
@@ -289,6 +324,346 @@ AxisAlignedBoundingBox AxisAlignedBoundingBox::FromLegacy(
+
+void OrientedBoundingBox::SetExtent(const core::Tensor &extent) {
+    core::AssertTensorDevice(extent, GetDevice());
+    core::AssertTensorShape(extent, {3});
+    if (extent.GetDtype() != GetDtype()) {
+        PrintWarningMessageFromTensor("extent", extent.GetDtype(), GetDtype());
+    }
+
+    if (extent.Min({0}).To(core::Float64).Item<double>() <= 0) {
+        utility::LogError(
+                "Invalid oriented bounding box. Please make sure the values of "
+                "extent are all positive.");
+    }
+
+    extent_ = extent.To(GetDtype());
+}
+
+void OrientedBoundingBox::SetRotation(const core::Tensor &rotation) {
+    core::AssertTensorDevice(rotation, GetDevice());
+    core::AssertTensorShape(rotation, {3, 3});
+    if (rotation.GetDtype() != GetDtype()) {
+        PrintWarningMessageFromTensor("rotation", rotation.GetDtype(),
+                                      GetDtype());
+    }
+
+    if (!rotation.T().AllClose(rotation.Inverse(), 1e-5, 1e-5)) {
+        utility::LogWarning(
+                "Invalid oriented bounding box. Please make sure the rotation "
+                "matrix is orthogonal.");
+    } else {
+        rotation_ = rotation.To(GetDtype());
+    }
+}
+
+void OrientedBoundingBox::SetColor(const core::Tensor &color) {
+    core::AssertTensorDevice(color, GetDevice());
+    core::AssertTensorShape(color, {3});
+    if (color.Max({0}).To(core::Float64).Item<double>() > 1.0 ||
+        color.Min({0}).To(core::Float64).Item<double>() < 0.0) {
+        utility::LogError(
+                "The color must be in the range [0, 1], but for range [{}, "
+                "{}].",
+                color.Min({0}).To(core::Float64).Item<double>(),
+                color.Max({0}).To(core::Float64).Item<double>());
+    }
+
+    color_ = color.To(GetDtype());
+}
+
+core::Tensor OrientedBoundingBox::GetMinBound() const {
+    return GetBoxPoints().Min({0});
+}
+
+core::Tensor OrientedBoundingBox::GetMaxBound() const {
+    return GetBoxPoints().Max({0});
+}
+
+core::Tensor OrientedBoundingBox::GetBoxPoints() const {
+    const core::Tensor x_axis =
+            rotation_.GetItem(
+                    {core::TensorKey::Slice(core::None, core::None, core::None),
+                     core::TensorKey::Index(0)}) *
+            extent_.GetItem({core::TensorKey::Index(0)}) * 0.5;
+    const core::Tensor y_axis =
+            rotation_.GetItem(
+                    {core::TensorKey::Slice(core::None, core::None, core::None),
+                     core::TensorKey::Index(1)}) *
+            extent_.GetItem({core::TensorKey::Index(1)}) * 0.5;
+    const core::Tensor z_axis =
+            rotation_.GetItem(
+                    {core::TensorKey::Slice(core::None, core::None, core::None),
+                     core::TensorKey::Index(2)}) *
+            extent_.GetItem({core::TensorKey::Index(2)}) * 0.5;
+
+    core::Tensor points = core::Tensor::Zeros({8, 3}, GetDtype(), GetDevice());
+    points.SetItem(core::TensorKey::Index(0),
```

Comment:
To investigate a cleaner initialization possibility.

Reply:
How about the following way:

```cpp
const t::geometry::AxisAlignedBoundingBox aabb(GetExtent() * -0.5, GetExtent() * 0.5);
return aabb.GetBoxPoints().Matmul(GetRotation()).Add(GetCenter());
 ```